src/app/context/AccountContext.tsx

```diff
@@ -32,15 +33,37 @@ interface AccountContextType {
    */
   setAccountId: (id: string) => void;
   /**
-   * Fetch the additional account info: alias/balance and update account
+   * revalidation (mark the data as expired and trigger a refetch) for the resource
    */
-  fetchAccountInfo: (options?: {
-    accountId?: string;
-  }) => Promise<AccountInfo | undefined>;
+  refetchAccountInfo: () => Promise<void>;
 }
 
 const AccountContext = createContext({} as AccountContextType);
 
+export const getAccountInfoKey = (
```

Comment:
@rolznz `getAccountInfoKey` and `buildAccountInfo` are also used in TestConnection. Should I create a `app/utils/accounts.ts` function and move them there?

Reply:
I think it would be better to make a reusable hook e.g. src/app/hooks/useSelectedAccount and put all this logic in one place. I think it might just need an accountId parameter?

Then in the TestConnection and AccountContext you can just use the hook. This would also make AccountContext simpler too. What do you think?

Reply:
I already used to have this abstracted as "useAccountInfoCached" in "api.ts" - then you mentioned to simplify things and call "useSWR" directly in the AccountContext. I guess it was a misunderstanding then and I also think it makes sense, so I refactor to abstract it and move it to the suggested place.

Reply:
@lujakob sorry about that.

Reply:
No worries.